### PR TITLE
Testnet Command

### PR DESCRIPTION
**Pull request by araskachoi**
- change tendermint version v0.33.8 -> v0.33.7
- change tm-db version v0.6.1 -> v0.5.1

- include testnet command
`aragonchaind testnet`
- change token denom to "ara"

### CODE REVIEW

File: client/testnet.go [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+
+		// save private key seed words
+		if err := writeFile(fmt.Sprintf("%v.json", "key_seed"), clientDir, cliPrint); err != nil {
+			return err
+		}
+
+		accTokens := sdk.TokensFromConsensusPower(1000)
+		accStakingTokens := sdk.TokensFromConsensusPower(5000)
+		coins := sdk.NewCoins(
+			sdk.NewCoin(sdk.DefaultBondDenom, accTokens),
```
Comment (fedekunze) [inline client/testnet.go:211]: the staking token needs to be changed to `"ara"`. Can you create a constant for that?

File: client/testnet.go [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+	}
+
+	cmd.Flags().Int(flagNumValidators, 4, "Number of validators to initialize the testnet with")
+	cmd.Flags().StringP(flagOutputDir, "o", "./build", "Directory to store initialization data for the testnet")
+	cmd.Flags().String(flagNodeDirPrefix, "node", "Prefix the directory name for each node with (node results in node0, node1, ...)")
+	cmd.Flags().String(flagNodeDaemonHome, "aragonchaind", "Home directory of the node's daemon configuration")
+	cmd.Flags().String(flagNodeCLIHome, "aragonchaincli", "Home directory of the node's cli configuration")
+	cmd.Flags().String(flagStartingIPAddress, "192.168.0.1", "Starting IP address (192.168.0.1 results in persistent peers list ID0@192.168.0.1:46656, ID1@192.168.0.2:46656, ...)")
+	cmd.Flags().String(flags.FlagChainID, "", "genesis file chain-id, if left blank will be randomly created")
+	cmd.Flags().String(server.FlagMinGasPrices, fmt.Sprintf("0.000006%s", types.DenomDefault), "Minimum gas prices to accept for transactions; All fees in a tx must meet this minimum (e.g. 0.01photon,0.001stake)")
```
Comment (fedekunze) [inline client/testnet.go:96]: ditto ara with constant

File: client/testnet.go [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+		if err != nil {
+			return err
+		}
+
+		// save private key seed words
+		if err := writeFile(fmt.Sprintf("%v.json", "key_seed"), clientDir, cliPrint); err != nil {
+			return err
+		}
+
+		accTokens := sdk.TokensFromConsensusPower(1000)
```
Comment (fedekunze) [inline client/testnet.go:208]: I think they only wanted one token denomination for the testnet

Comment (fedekunze) [inline client/testnet.go:208]: I think they only wanted one token denomination for the testnet
```suggestion
```

File: client/testnet.go (hunk 1 above)
Comment (fedekunze) [reply client/testnet.go:211]: sorry this applies to the staking token below

File: client/testnet.go [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+		genAccounts = append(genAccounts, types.EthAccount{
+			BaseAccount: authtypes.NewBaseAccount(addr, coins, nil, 0, 0),
+			CodeHash:    ethcrypto.Keccak256(nil),
+		})
+
+		valTokens := sdk.TokensFromConsensusPower(100)
+		msg := stakingtypes.NewMsgCreateValidator(
+			sdk.ValAddress(addr),
+			valPubKeys[i],
+			sdk.NewCoin(types.DenomDefault, valTokens),
```
Comment (fedekunze) [inline client/testnet.go:224]: ditto

**Comment by araskachoi** [comment]
i think g2g @fedekunze! approve when you have a chance

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# Install ca-certificates
+RUN apk add --update ca-certificates
+WORKDIR /root
+
+# Copy over binaries from the build-env 
+COPY --from=build-env /go/src/github.com/aragon/aragon-chain/build/aragonchaind /usr/bin/aragonchaind
+COPY --from=build-env /go/src/github.com/aragon/aragon-chain/build/aragonchaincli /usr/bin/aragonchaincli
+
+# Run ethermintd by default
+CMD ["aragonchaind"]
```
Comment (promaty) [inline Dockerfile:31]: This is only for the node but no JSON rpc right?
For our deployment we need containers exposing 2 ports for api and p2p discovery

Comment (araskachoi) [reply Dockerfile]: exposed the ports in the dockerfile. the `aragonchaincli` is the binary for the json rpc rest-server. 

I changed the entrypoint to bash shell so you can pass the command to run that binary.

**Review by fedekunze — APPROVED** [review]
I will approve and merge this PR since the changes from the testnet command are completed. I also think we should address the remaining fixes and consensus failures in a follow-up PR. 

cc: @sohkai @noot

File: Makefile [hunk 6]
```diff
@@ -236,7 +252,7 @@ format:
 	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -name '*.pb.go' | xargs goimports -w -local github.com/tendermint
 	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -name '*.pb.go' | xargs goimports -w -local github.com/ethereum/go-ethereum
 	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -name '*.pb.go' | xargs goimports -w -local github.com/cosmos/cosmos-sdk
-	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -name '*.pb.go' | xargs goimports -w -local github.com/cosmos/ethermint
+	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -name '*.pb.go' | xargs goimports -w -local github.com/cosmos/aragonchain
```
Comment (fedekunze) [inline Makefile:255]:
```suggestion
	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -name '*.pb.go' | xargs goimports -w -local github.com/cosmos/ethermint
```

**Pull request merged**